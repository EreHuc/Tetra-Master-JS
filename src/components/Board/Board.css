/* FIXME: Make it work with various grid sizes */

.board {
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(3, 100px);
  grid-template-rows: 100px 100px 100px;
  grid-auto-flow: column;
}

.board-cell {
  background: lightgrey;
  display: flex;
  justify-content: center;
  flex-direction: column;
  text-align: center;
}
