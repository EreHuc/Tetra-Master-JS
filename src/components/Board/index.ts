export * from "./Board";
export { default } from "./Board.container";
