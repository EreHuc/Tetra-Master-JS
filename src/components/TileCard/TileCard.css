.tile-card {
  width: 64px;
  height: 64px;
  border: 2px solid transparent;
}

.tile-card.selected {
  border-color: #000;
}

.tile-type-1 {
  background: lightblue;
}

.tile-type-2 {
  background: lightcoral
}

.tile-type-3 {
  background: lightyellow
}

.tile-type-4 {
  background: lightgreen
}
