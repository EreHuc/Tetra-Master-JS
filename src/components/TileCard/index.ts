export { default } from "./TileCard.container";
export * from "./TileCard";
