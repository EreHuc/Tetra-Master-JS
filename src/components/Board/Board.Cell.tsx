import * as React from "react";
import { defaultProps } from "recompose";

import { Cell, Vector2 } from "../../models";
import TileCard from "../TileCard";

type CellProps = {
  position: Vector2;
  cell: Cell;
  onClick: (position: Vector2) => void;
};

const onClickWithPosition = (onClick, position) => clickEvent => {
  onClick(position);
};

const BoardCell: React.SFC<CellProps> = ({ position, cell, onClick }) => (
  <div
    className="board-cell"
    role="button"
    onClick={onClickWithPosition(onClick, position)}
  >
    {cell.tileId && <TileCard tileId={cell.tileId} />}
  </div>
);

export const EnhancedBoardCell = defaultProps({
  onClick: _ => {},
})(BoardCell);
