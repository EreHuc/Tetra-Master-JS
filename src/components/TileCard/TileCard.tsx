import * as React from "react";

import { Tile } from "../../models";
import "./TileCard.css";

type TileProps = {
  tile: Tile;
  focused?: boolean;
  selected?: boolean;
  onMouseEnter?: React.MouseEventHandler;
  onMouseLeave?: React.MouseEventHandler;
  onClick?: React.MouseEventHandler;
};

export const TileCard: React.SFC<TileProps> = ({
  tile,
  selected,
  focused,
  onMouseEnter,
  onMouseLeave,
  onClick,
}) => (
  <button
    className={`tile-card tile-type-${tile.typeId} ${selected && "selected"}`}
    onMouseEnter={onMouseEnter}
    onMouseLeave={onMouseLeave}
    onClick={onClick}
  >
    {focused && `tile ${tile.id}`}
  </button>
);
