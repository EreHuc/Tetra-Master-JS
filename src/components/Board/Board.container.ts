import { connect } from "react-redux";

import { randomId } from "../../lib/randomId";
import { getBoardGrid, placeTile } from "../../store";
import { Board } from "./Board";

const mapStateToProps = state => ({
  grid: getBoardGrid(state),
});

const mapDispatchToProps = dispatch => ({
  onCellClick: position => {
    // FIXME: Use the currentPlayerId and selectedTileId instead of random ids.
    dispatch(placeTile(randomId(1, 2), randomId(1, 2), position));
  },
});

const enhance = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default enhance(Board);
