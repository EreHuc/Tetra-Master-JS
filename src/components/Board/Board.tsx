import * as React from "react";

import { Grid, Vector2 } from "../../models";
import { EnhancedBoardCell } from "./Board.Cell";
import "./Board.css";

type BoardProps = {
  grid: Grid;
  onCellClick: (position: Vector2) => void;
};

export const Board: React.SFC<BoardProps> = ({ grid, onCellClick }) => (
  <div className="board">
    <div className="grid">
      {grid.map((row, x) =>
        row.map((cell, y) => (
          <EnhancedBoardCell
            key={`${x}-${y}`}
            cell={cell}
            position={new Vector2(x, y)}
            onClick={onCellClick}
          />
        )),
      )}
    </div>
  </div>
);
