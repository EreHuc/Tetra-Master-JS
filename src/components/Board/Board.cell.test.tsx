import { shallow } from "enzyme";
import * as React from "react";

import TileCard from "../TileCard";
import { EnhancedBoardCell } from "./Board.Cell";

const mockCell = { playerId: "1", tileId: "1" };

describe("EnhancedBoardCell", () => {
  it("should display a tile with the correct tileId", () => {
    const wrapper = shallow(
      <EnhancedBoardCell cell={mockCell} position={{ x: 1, y: 1 }} />,
    ).dive();

    expect(wrapper.find(TileCard)).toHaveLength(1);
    expect(wrapper.find(TileCard).get(0).props.tileId).toBe("1");
    expect(wrapper).toMatchSnapshot();
  });

  it("should react to clicks", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <EnhancedBoardCell
        cell={mockCell}
        position={{ x: 1, y: 1 }}
        onClick={onClick}
      />,
    ).dive();

    wrapper.find("div").simulate("click");

    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith({ x: 1, y: 1 });
    expect(wrapper).toMatchSnapshot();
  });
});
