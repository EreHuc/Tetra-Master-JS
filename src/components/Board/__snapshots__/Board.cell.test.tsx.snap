// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedBoardCell should display a tile with the correct tileId 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoardCell
    cell={
      Object {
        "playerId": "1",
        "tileId": "1",
      }
    }
    onClick={[Function]}
    position={
      Object {
        "x": 1,
        "y": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(TileCard)
        tileId="1"
      />,
      "className": "board-cell",
      "onClick": [Function],
      "role": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "tileId": "1",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(TileCard)
          tileId="1"
        />,
        "className": "board-cell",
        "onClick": [Function],
        "role": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "tileId": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`EnhancedBoardCell should react to clicks 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoardCell
    cell={
      Object {
        "playerId": "1",
        "tileId": "1",
      }
    }
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "x": 1,
              "y": 1,
            },
          ],
        ],
      }
    }
    position={
      Object {
        "x": 1,
        "y": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(TileCard)
        tileId="1"
      />,
      "className": "board-cell",
      "onClick": [Function],
      "role": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "tileId": "1",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(TileCard)
          tileId="1"
        />,
        "className": "board-cell",
        "onClick": [Function],
        "role": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "tileId": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
